api/fasit.go

```diff
@@ -61,6 +63,53 @@ func (fasit FasitClient) GetResources(resourcesRequests []ResourceRequest, envir
 	return resources, nil
 }
 
+func (fasit FasitClient) createOrUpdateFasitResources(exposedResourceIdsRef *[]int, resources []ExposedResource, hostname, environment, application, zone string) (error) {
+	exposedResourceIds := *exposedResourceIdsRef
+	for _, resource := range resources {
+		var request = ResourceRequest{Alias: resource.Alias, ResourceType: resource.ResourceType}
+		existingResource, err := fasit.getResource(request, environment, application, zone)
+
+		if err != nil {
+			// Create new resource if none was found
+			createdResourceId, err := fasit.createResource(resource, environment, application, zone)
+			if err != nil {
+				return fmt.Errorf("Failed creating resource: %s, %s", resource.Alias, err)
+			}
+
+			exposedResourceIds = append(exposedResourceIds, createdResourceId)
+			continue
+			//} else if resourceHasChanged(existingResource, resource) {
+			//	// updateResource
+			//	if err != nil {
+			//		return exposedResourceIds, fmt.Errorf("Failed updating resource: %s, %s", resource.Alias, err)
+			//	}
+			//	exposedResourceIds = append(exposedResourceIds, updatedResource.id)
+			//	continue
+		}
+
+		// UnchangedResource
+		exposedResourceIds = append(exposedResourceIds, existingResource.id)
+	}
+	return nil
+}
+
+func getResourceIds(usedResources []UsedResource) (usedResourceIds []int) {
+	for _, resource := range usedResources {
+		usedResourceIds = append(usedResourceIds, resource.Id)
+	}
+	return usedResourceIds
+}
+
+func environmentExistsInFasit(fasitUrl string, deploymentRequest NaisDeploymentRequest) (error) {
+	fasit := FasitClient{fasitUrl, deploymentRequest.Username, deploymentRequest.Password}
```

Comment:
maybe initialize fasitclient only once pr file, then use  dep injection

Reply:
Looks like  fasit requires basic auth, so a fasit client pr request is needed.